tests/stub/routing.py

```diff
@@ -2222,6 +2234,30 @@ def work(tx):
         self._routingServer3.done()
         self._readServer1.done()
         self.assertEqual([[1]], sequences)
+        if len(resolver_calls) == 1:
+            # driver that calls resolver function only on initial router address
+            self.assertEqual(resolver_calls.keys(),
+                             {self._routingServer1.address})
+            # depending on whether the resolve result is treated equally to a
+            # RT table entry or is discarded after an RT has been retrieved
+            # successfully.
+            self.assertIn(resolver_calls[self._routingServer1.address], {1, 2})
```

Comment:
Out if interest, which drivers would get `1` and `2` here?

Reply:
.net, js, and go, skip this test, only java reaches this branch and calls the resolver twice for `_routingServer1`. However, if we decide to switch the the routing/resolving algorithm I proposed, this will be reduced to a single call.

Reply:
Actually, no. It wouldn't. It should always be two. I'll change that.